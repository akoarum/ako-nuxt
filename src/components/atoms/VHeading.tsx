import Vue, { CreateElement, VNode } from 'vue'

export default Vue.extend({
  props: {
    level: { type: Number, default: 2 },
    visual: { type: Number, default: 2 }
  },
  render(h: CreateElement): VNode {
    const Tag: string = `h${this.level}`
    const classes: string[] = ['vHeading']
    const content = this.$slots.default ? this.$slots.default[0].text : ''

    classes.push(`lv${this.visual}`)

    return <h2 { ...{ style: styles[`lv${this.visual}`] }}>{content}</h2>
  }
})

const styles: { [key: string]: any } = {
  lv2: {
    fontSize: '2.3rem',
    lineHeight: '1.4'
  },
  lv3: {
    fontSize: '2rem',
    lineHeight: '1.5'
  },
  lv4: {
    fontSize: '1.8rem',
    lineHeight: '1.5'
  }
}
