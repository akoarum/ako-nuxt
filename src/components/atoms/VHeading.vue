<script>
export default {
  props: {
    level: { type: Number, default: 2 },
    visual: { type: Number, default: 2 }
  },
  render(h) {
    const Tag = `h${this.level}`
    return <Tag class={`vHeading${this.visual}`}>{this.$slots.default}</Tag>
  }
}
</script>

<style lang="scss" scoped>
.vHeading2 {
  @include media-min {
    font-size: 2.3rem;
    line-height: 1.4;
  }
  @include media-max {
    font-size: 2.2rem;
    line-height: 1.5;
  }
}

.vHeading3 {
  font-size: 2rem;
  line-height: 1.5;
}

.vHeading4 {
  font-size: 1.8rem;
  line-height: 1.5;
}
</style>
