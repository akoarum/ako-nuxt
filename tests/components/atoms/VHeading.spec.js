import { shallowMount } from '@vue/test-utils'
import VHeading from '~/components/atoms/VHeading'

describe('VHeading', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(VHeading, {
      propsData: {}
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })
    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
