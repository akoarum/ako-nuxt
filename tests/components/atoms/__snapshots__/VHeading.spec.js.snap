// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VHeading initialized snapshot 1`] = `
<h2
  class="vHeading2"
/>
`;
